[Yang Fan] ip

nus-sg opened the pull request:
PRs Task

nus-sg committed:
Level 0 commit. Set up IDE and Source Tree

nus-sg committed:
Complete "level 0"

nus-sg committed:
Completed "level 1"

nus-sg committed:
Merge branch 'master' of https://github.com/fanceso/ip

nus-sg committed:
Level 2 task completed. Added "listContent" and "addContent" method

nus-sg committed:
Minor edit for Level-2 task

nus-sg committed:
Completed Level 3
Added in "Task.class" for identify task checking to status TICK or CROSS in Unix symbol.

nus-sg committed:
Follow the Coding Standard

nus-sg committed:
Level 4. ToDos, Events, Deadlines
Add support for tracking three types of tasks:

1. ToDos: tasks without any date/time attached to it e.g., visit new theme park
2. Deadlines: tasks that need to be done before a specific date/time e.g., submit report by 11/10/2019 5pm
3. Events: tasks that start at a specific time and ends at a specific time e.g., team project meeting on 2/10/2019 2-4pm

nus-sg committed:
Update README.md

weisiong24 commented on src/main/java/Duke.java:
```diff
@@ -63,48 +117,53 @@ private static void exitProgram() {
-        taskIdentify();
+        String listContent = "";
+        int i;
+        if (taskCount == 0) {
```
Nice! You add conditions and behaviors where there is no task done

riazaham commented on src/main/java/Duke.java:
```diff
@@ -1,18 +1,49 @@
+        switch (userCommand) {
+        case COMMAND_EXIT_STRING:
+            exitProgram();
+        default:
```
Missing a 'break' statement

riazaham reviewed (commented):
CS2113T Tutorial PR exercise

nus-sg committed:
Level 5: Error Handling
Text-UI-Testing: Using "Input.txt" with "ACTUAL.TXT" to compare the expected output with "EXPECTED.TXT"

nus-sg committed:
A-Packages

nus-sg committed:
Level-5 Branch: With A-packages, A-TextUiTesting.

nus-sg committed:
Branch Demo

nus-sg committed:
Merge branch 'branch-Level-5'

nus-sg committed:
Level 6.
Delete Implemented.
Changing array to arrayList

nus-sg committed:
Level 7. Saving into file

nus-sg committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/duke/Duke.java

Solved conflicts

nus-sg committed:
Added  StandardCharsets.UTF_8

nus-sg committed:
Fixed utf-8 reading problem

nus-sg committed:
Created Message Package

JustinnT commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,341 @@
+public class Duke {
+    private static final Scanner SCANNER = new Scanner(System.in);
+    private static final String FILE_PATH_NAME = System.getProperty("user.dir") + "\\data-folder";
+    private static final String FILE_NAME = FILE_PATH_NAME + "\\data.txt";
```
Good job, the strings should be a constant since it will not be changed.

JustinnT commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,341 @@
+        // create directory if does not exist
+        if (!theDir.exists()) {
+            System.out.println("Creating new directory: " + theDir.getName());
+            boolean result = false;
```
Perhaps can change the boolean variable name to sound more boolean like. This is because having a non-boolean like name may result in lower readability. Example: "isSuccessful"

JustinnT commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,341 @@
+    public static String taskContent = "";
+    private static int taskCount = 0;
+    private static boolean loaded = false;
+    private static boolean autoSaveMode = false;
```
Should the booleans have more boolean like names?

JustinnT commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,341 @@
+            }
+        } catch (InvalidCommandException exception) {
+            switch (exception.errorCode) {
+            case "common":
```
I like how your above switch cases are expressed in constants. Maybe can consider doing it here as well.

JustinnT commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,341 @@
+                    } else {
+                        completed = true;
+                        content = s.nextLine();
+                        int dividePoint1 = content.indexOf(".[");
+                        int dividePoint2 = content.indexOf("] ");
+                        char taskType = content.charAt(dividePoint1 + 2);
+                        String taskDetails = content.substring(dividePoint2 + 2);
+                        String taskIsDone = content.substring(dividePoint1 + 5, dividePoint2);
```
Should we apply SLAP here to make the code more readable? Perhaps something like taskType = getTaskType();

JustinnT reviewed (commented):
Good job, have to take note of the boolean names

nus-sg committed:
A-MoreOOP

nus-sg committed:
A-MoreOOP Version updated.

nus-sg committed:
Level 8: Date time record with the aid of DateTimeFormatter

nus-sg committed:
Find function implement using Stream method.

nus-sg committed:
A-JavaDoc Completed

nus-sg committed:
Merge pull request #1 from fanceso/A-JavaDoc

A java doc completed

nus-sg committed:
Duke A-UserGuide: User Guide
Adds a user guide for duke

nus-sg committed:
Minor fix for user guide

nus-sg committed:
User guide

nus-sg committed:
Final touch up on UG

nus-sg committed:
A-UserGuide

nus-sg committed:
Revert "Final touch up on UG"

This reverts commit 89701cfdf74c06c83af173c246f72cb563e408bd.

nus-sg committed:
A-UserGuide

nus-sg committed:
v0..2 release

nus-sg committed:
Set theme jekyll-theme-minimal

nus-sg committed:
Set theme jekyll-theme-minimal

nus-sg committed:
Set theme jekyll-theme-minimal

nus-sg committed:
Set theme jekyll-theme-slate

nus-sg committed:
Update for github pages

nus-sg committed:
Merge branch 'master' of https://github.com/fanceso/ip

nus-sg committed:
Final Updates on UG and Pages
